cypress/integration/BounceRulesContainer/bounce_rules.spec.js

```diff
@@ -16,23 +34,24 @@ describe("Bounce Rules Page", () => {
   });
 
   it("should create a bounce rule", () => {
-    BounceRulesPage.createRuleButton.click();
-    BounceRulesPage.fillForm(
-      "1",
-      "cypressTest",
-      "528",
-      "testDescription",
-      "492",
-      "testRegex"
-    );
-    BounceRulesPage.confirmModal.should("be.visible");
-    BounceRulesPage.confirmationSubmit.click();
-    BounceRulesPage.testRule.should("be.visible");
+    BounceRulesPage.deleteBounceRuleAPI(testCreateRule).then(() => {
+      BounceRulesPage.createBounceRuleUI(testCreateRule).then(() => {
+        BounceRulesPage.confirmModal.should("be.visible");
+        BounceRulesPage.confirmationSubmit.click().then(() => {
+          BounceRulesPage.createdCreateRuleAPI.should("be.visible").then(() => {
+            BounceRulesPage.deleteBounceRuleAPI(testCreateRule);
+          });
+        });
+      });
+    });
   });
 
   it("should delete a bounce rule", () => {
-    BounceRulesPage.testRule.click();
-    BounceRulesPage.deleteConfirmationConfirm.click();
-    BounceRulesPage.testRule.should("not.be.visible");
+    BounceRulesPage.createBounceRuleAPI(testDeleteRule).then(id => {
+      BounceRulesPage.createdRuleButton(id).click();
```

Comment:
Same here let's encapsulate all the delete steps (two clicks) inside a function and do one .then() and make the final assertion inside of that